drivers/stepper/Kconfig.step_dir

```diff
@@ -0,0 +1,7 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 Fabian Blatz
+# SPDX-License-Identifier: Apache-2.0
+
+config STEP_DIR_STEPPER
+	bool
+	help
+	  Enable library used for step direction stepper drivers.
```

Comment:
Reason for the symbol is that in the CMakeLists you only want to include the source file if its used. The Kconfig symbol is the most convenient way of doing that.

Reply:
We need this kconfig variable for sure. But since this is a common property to lot of drivers, it can be placed in drivers/stepper/kconfig.

At this point I don't think we need a dedicated file for one variable. If need be we can always refactor Lateron like we did with tmc5041.

Reply:
Ah, misunderstood you then :^), moved it